### DISCUSSION TITLE
Create CA cert with custom CN, OU, O, L, S, C

### DISCUSSION THREAD

**Discussion by patsevanton** [smallstep/certificates#1735] [Q&A] [answered]
Hello! Thanks for smallstep!
I read https://github.com/smallstep/certificates/blob/master/examples/docker/ca/pki/config/ca.json
I read https://github.com/smallstep/certificates/discussions/379
but i not found answer.
How create CA cert with custom CN, OU, O, L, S, C ? (i use docker-compose)

**Comment by hslatman** [comment, marked as answer]
Hey @patsevanton,

When you do `step ca init`, some sane defaults are picked. To keep things simple, we don't provide all options, and thus use some defaults for the contents of the root and intermediate certificate. It's possible to create your own root and intermediate certificates using `step certificate create --template /path/to/template`. You can use the template to customize the contents. You'll need to take care of the proper settings in the template, so you'll likely want them to be similar to what's in https://github.com/smallstep/crypto/blob/master/x509util/templates.go#L193-L214. Then after creating the root and intermediate (keys and certs) in this way, you can use the instructions from https://smallstep.com/docs/tutorials/intermediate-ca-new-ca/index.html#the-medium-way to initialize your CA with your own root and intermediate.
